From 3b4c07005c330db0b31615c4e458fc2219aab767 Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Tue, 22 Apr 2014 13:42:05 +0800
Subject: [PATCH 446/449] ENGR00306106-2 Add new command to control SOF
 stuffing

VPU MJPEG encoder pads stuffing zero bytes to the end
of SOF fields by default. User may use new command
ENC_ENABLE_SOF_STUFF to enable or disable SOF stuffing.

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 test/mxc_vpu_test/enc.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/test/mxc_vpu_test/enc.c b/test/mxc_vpu_test/enc.c
index dc5e085..fcfabe7 100644
--- a/test/mxc_vpu_test/enc.c
+++ b/test/mxc_vpu_test/enc.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004-2013 Freescale Semiconductor, Inc.
+ * Copyright 2004-2014 Freescale Semiconductor, Inc.
  *
  * Copyright (c) 2006, Chips & Media.  All rights reserved.
  */
@@ -316,7 +316,9 @@ put_mp4header:
 		if (enc->qMatTable)
 			free(enc->qMatTable);
 		if (cpu_is_mx6x()) {
+			int enableSofStuffing = 1;
 			EncParamSet enchdr_param = {0};
+			vpu_EncGiveCommand(handle, ENC_ENABLE_SOF_STUFF, &enableSofStuffing);
 			enchdr_param.size = STREAM_BUF_SIZE;
 			enchdr_param.pParaSet = malloc(STREAM_BUF_SIZE);
 			if (enchdr_param.pParaSet) {
-- 
1.8.0

